README.md

```diff
@@ -933,6 +933,92 @@ _syscall.number()_ returns the system call number referenced by the given `name`
+
+#### `netfilter.match(match_func), nf:match(match_func)`
+
+_netfilter.match(match_func)_ registers the specified `match_func`. The `match_func` is a function that will be called when a packet is processed by the xtable. The function must return boolean value indicating whether the packet should be accepted or dropped.
+
+#### `netfilter.matches(matches)`
+
+_netfilter.matches(matches)_ registers multiple `matches`. The `matches` is a table that contains the name of the filter object and the corresponding match functions that will be called when a packet is processed by the xtable. The function must return boolean value indicating whether the packet should be accepted or dropped.
+
+#### `netfilter.target(target_func), nf:target(target_func)`
+
+_netfilter.target(target_func)_ registers the specified `target_func`. The `target_func` is a function that will be called when a packet is processed by the xtable. The function must return a boolean value indicating whether the packet should be accepted or dropped.
+
+#### `netfilter.targets(targets)`
+
+_netfilter.targets(targets)_ registers multiple `targets`. The `targets` is a table that contains the name of the filter object and the corresponding target functions that will be called when a packet is processed by the xtable. The function must return a boolean value indicating whether the packet should be accepted or dropped.
+
+#### `netfilter.hook(ops)`
+
+_netfilter.hook(ops)_ registers a netfilter hook with the specified `ops`. The `ops` is a table with the following fields:
+
+* `hook`: a callback function that returns a value one of the following values:
+  * `NF_ACCEPT`: accept the packet.
+  * `NF_DROP`: drop the packet.
+  * `NF_STOLEN`: the packet is stolen by the hook.
+  * `NF_QUEUE`: the packet is queued.
+  * `NF_REPEAT`: the packet is repeated.
+  * `NF_STOP`: stop the traversal.
+
+* `priority`: the priority of the hook.
+* `pf`: the protocol family. This is one of :
+  * `NFPROTO_UNSPEC` (unspecified).
+  * `NFPROTO_INET` (IPv4).
+  * `NFPROTO_IPV4` (IPv4).
+  * `NFPROTO_ARP` (ARP).
+  * `NFPROTO_NETDEV` (network device).
+  * `NFPROTO_BRIDGE` (bridge).
+  * `NFPROTO_IPV6` (IPv6).
+* `hooknum`: the hook number. This is one of :
+  * `NF_INET_PRE_ROUTING` (pre-routing).
+  * `NF_INET_LOCAL_IN` (local input).
+  * `NF_INET_FORWARD` (forward).
+  * `NF_INET_LOCAL_OUT` (local output).
+  * `NF_INET_POST_ROUTING` (post-routing).
+  * `NF_INET_NUMHOOKS` (number of hooks).
+
+#### `netfilter.hooks(hooks)`
+
+_netfilter.hooks(hooks)_ registers multiple `hooks`. The `hooks` is a table that contains the ops as described above.
+
+#### Example Script
+
+```lua
+local nf = require("netfilter")
+
+function drop_packet(skb)
+	print("drop_packet called")
+	return false
```

Comment:
shouldn't use `NF_DROP` here?

Reply:
NF_DROP can be used, but it may give the idea that `NF_*` values are to be used, however, only boolean values can be used (NF_DROP and NF_ACCEPT qualify this).

Reply:
why it can't be used? if I recall correctly nflua uses it.

Reply:
Yes, `NF_*` can be used for `target` but not for `match` (since callback should return a bool) , so I will create a different function for target callback. nflua also uses this for targets only.